### ISSUE TITLE
not recognizing IPv6 addresses?

### ISSUE THREAD

**Issue by c0b** [pashky/restclient.el#131]
since IPv6 addresses becoming increasing popular, we need to deal with some services running on IPv6, a browser can visit http://[::1]:3000/some-url without any problems, but this restclient seems not recognizing it:

```console
GET http://[::1]:3000/some-url             # C-c C-c to send query

Contacting host: [::1]:3000
[::1]/3000 Temporary failure in name resolution
```

**Comment by pashky** [comment]
The fact that you're getting that far means issue isn't in restclient.el. I'd suggest researching if url.el and your build of emacs are capable of connecting to IPv6.

Have you tried using localhost resolving to ::1 first? Did it work? There were known issues with IPv6 in the past (see #43), but this seems different.

**Comment by pashky** [comment]
Ok, I checked it out of curiosity. Handling IPv6 literals in host part of URLs is broken in emacs and should be fixed on many different levels. 

So either use hostnames or give up, IPv6 is still quite dead anyway :)

**Comment by c0b** [comment]
no, that doesn't seem a good excuse, this is just latest Ubuntu's default emacs,

I believe it's more of an url parsing issue, need to parse out the address from `[::1]` and turn into a valid IPv6 address, and pass to low level networking API

like the curl can also parse it correctly:

```console
➸ curl -v http://[::1]:3000/some-url
*   Trying ::1...
* Connected to ::1 (::1) port 3000 (#0)
> GET /some-url HTTP/1.1
> Host: [::1]:3000
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 404 Not Found
< Content-Type: application/json
< Date: Fri, 22 Jul 2016 19:45:44 GMT
< Connection: keep-alive
< Transfer-Encoding: chunked
< 
{
  "message": "/some-url does not exist",
...
```

**Comment by pashky** [comment]
`M-x report-emacs-bug`
